desktop/ui/src/main/java/org/datacleaner/util/RemoteServersUtils.java

```diff
@@ -0,0 +1,106 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.util;
+
+import java.util.List;
+
+import org.datacleaner.configuration.DataCleanerConfiguration;
+import org.datacleaner.configuration.RemoteServerData;
+import org.datacleaner.configuration.RemoteServerDataImpl;
+import org.datacleaner.descriptors.CompositeDescriptorProvider;
+import org.datacleaner.descriptors.RemoteDescriptorProvider;
+import org.datacleaner.descriptors.RemoteDescriptorProviderImpl;
+import org.datacleaner.restclient.ComponentRESTClient;
+
+/**
+ * Utilities for better work with remote servers.
+ */
+public class RemoteServersUtils {
+    final private DataCleanerConfiguration _configuration;
+
+    public RemoteServersUtils(final DataCleanerConfiguration configuration) {
+        _configuration = configuration;
+    }
+
+    public RemoteServerData getServerConfig(String serverName) {
+        if (serverName == null) {
+            return null;
+        }
+        List<RemoteServerData> serverList =
+                _configuration.getEnvironment().getRemoteServerConfiguration().getServerList();
+        for (RemoteServerData remoteServerData : serverList) {
+            String configServerName = remoteServerData.getServerName();
+            if (configServerName == null) {
+                continue;
+            }
+            if (configServerName.toLowerCase().equals(serverName.toLowerCase())) {
+                return remoteServerData;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * It creates new RemoteServerData and Remote server provider
+     *
+     * @param serverName
+     * @param serverUrl
+     * @param userName
+     * @param password
+     */
+    public void createRemoteServer(String serverName, String serverUrl, String userName, String password) {
+        String url = serverUrl;
+        if (RemoteDescriptorProvider.DATACLOUD_SERVER_NAME.equals(serverName) && url == null) {
+            url = RemoteDescriptorProvider.DATACLOUD_URL;
+        }
+        List<RemoteServerData> serverList =
+                _configuration.getEnvironment().getRemoteServerConfiguration().getServerList();
+        RemoteServerData remoteServerData =
+                new RemoteServerDataImpl(url, serverName, userName, password);
+        serverList.add(remoteServerData);
+
+        final CompositeDescriptorProvider descriptorProvider =
+                (CompositeDescriptorProvider) _configuration.getEnvironment().getDescriptorProvider();
```

Comment:
But a ```DataCleanerConfiguration``` can actually be created in other ways than reading conf.xml. And then by default it will not be a composite. That could be in simple testing scenarios, in DC embedded inside Pentaho, in DataHub or in dataimprover.com ... And anyways the DataCleanerConfiguration interface only returns the ```DescriptorProvider``` interface so that is the contract we have to live up to. The point just being: It's not safe to assume that DataCleanerConfiguration is within your control.